rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -649,7 +646,9 @@ def decode_transaction_hashes(
                 ignore_cache=ignore_cache,
                 delete_customized=delete_customized,
             )
-            events.extend(new_events)
+
+            if return_events:
```

Comment:
I would just create a different function that returns the events. No point in having it return empty list.